File: mid/Array/128. Longest Consecutive Sequence.cpp
// violent solutions
// Time : O(n*m)
// Space : O(n)

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> h(nums.begin(), nums.end());
        int ans = 0;
        
        for (int num : nums)
            if (!h.count(num - 1)){
                int l = 0;
                while (h.count(num++)) ++l;
                ans = max(ans, l);
            }
        return ans;
    }
};

File: mid/Array/128. Longest Consecutive Sequence.java
// Time : O(2N+N)
// Space : O(N)

class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> arr = new HashSet<>();
        for (int num : nums) {
            arr.add(num);
        }
        
        int maxx = 0;
        
        for (int i : nums) {
            int x = i - 1;
            int curr = 0;
            
            if (!arr.contains(x)) {
                while (arr.contains(x + 1)) {
                    curr += 1;
                    x += 1;
                }
                
                maxx = Math.max(maxx, curr);
            }
        }
        
        return maxx;
    }
}